src/index.ts

```diff
@@ -280,43 +251,49 @@ internals.noop = () => {}
  * @param {String} namespace
  * @return {Logger}
  */
-internals.syncLogger = function (logger, namespace) {
-    _.forOwn(logger, (value, key) => {
-        delete logger[key]
-    })
+export const syncLogger = (logger: Logger, namespace: string): Logger => {
+    for (const key in logger) {
+        delete logger[key as keyof Logger]
+    }
 
-    const enabledLevels = {}
-    internals.levels.forEach((level, idx) => {
-        if (level === 'none') return
-        if (!internals.isEnabled(namespace, idx)) {
-            enabledLevels[level] = false
-            logger[level] = internals.noop
-        } else {
-            enabledLevels[level] = true
-            logger[level] = function (contextId, message, data) {
-                internals.log(namespace, level, contextId, message, data)
+    const enabledLevels: Record<string, boolean> = {}
+    if (internals.levels) {
+        internals.levels.forEach((level, idx) => {
+            if (level === 'none') return
+            if (!internals.isEnabled?.(namespace, idx)) {
+                enabledLevels[level] = false
+                logger[level] = () => {}
+            } else {
+                enabledLevels[level] = true
+                logger[level] = (contextId: string, message: string, data?: Record<string, unknown>) => {
+                    log(namespace, level, contextId, message, data)
+                }
             }
-        }
-    })
+        })
 
-    logger.isLevelEnabled = (level) => enabledLevels[level]
+        logger.isLevelEnabled = (level) => enabledLevels[level]
+    }
 
     return logger
 }
 
 /**
- * Resync all loggers level functions to enable / disable them
+ * ReSync all loggers level functions to enable / disable them
  * This should be called when namespaces or levels are updated
  */
-internals.syncLoggers = function () {
-    _.forOwn(internals.loggers, (logger, namespace) => {
-        internals.syncLogger(logger, namespace)
-    })
+export const syncLoggers = () => {
+    for (const [namespace, logger] of Object.entries(internals.loggers)) {
+        syncLogger(logger, namespace)
+    }
 }
 
 /************* INIT *************/
 const namespaces = process.env.LOGS || '*'
-const logLevel = process.env.LOG_LEVEL || 'warn'
+const logLevel: LogLevel = (process.env.LOG_LEVEL as LogLevel) || 'warn'
+
+setNamespaces(namespaces)
+setLevel(logLevel)
 
-module.exports.setNamespaces(namespaces)
-module.exports.setLevel(logLevel)
+export * from './definitions'
+export { outputAdapters as outputs }
+export { outputAdapters, outputUtils, internals }
```

Comment:
```suggestion
export * from "./definitions"
export * as outputUtils from "./output_utils"
export * as outputAdapters from "./output_adapters"

export default {
    createLogger,
    setLevel,
    setNamespaces,
    setOutput,
    setGlobalContext,
    id
}
```

No need to expose `internals`

Reply:
Thank you so much for helping me to resolve the problem of `import Logger from ...` with this solution `export default`